TestApp/CrashesScreen.js

```diff
@@ -93,14 +104,103 @@ export default class CrashesScreen extends Component {
               Crash native code
             </Text>
           </TouchableOpacity>
+          <TouchableOpacity onPress={() => { this.dialogComponent.show(); }}>
+            <Text style={styles.button}>
+              Set text error attachment
+            </Text>
+          </TouchableOpacity>
+          <Text style={SharedStyles.enabledText}>{'Current value:'}{this.state.textAttachment}</Text>
+          <TouchableOpacity onPress={this.showFilePicker}>
+            <Text style={styles.button}>
+              Select image as binary error attachment
+            </Text>
+          </TouchableOpacity>
+          <Text style={SharedStyles.enabledText}>{'Current value:'}{this.state.binaryAttachment}</Text>
           <Text style={styles.lastSessionHeader}>Last session:</Text>
           <Text style={styles.lastSessionInfo}>
             {this.state.lastSessionStatus}
           </Text>
         </ScrollView>
+        {this.getTextAttachmentDialog()}
       </View>
     );
   }
+
+  getTextAttachmentDialog() {
+    return (
+      <DialogComponent
+        ref={(dialogComponent) => { this.dialogComponent = dialogComponent; }}
+        width={0.9}
+      >
+        <View>
+          <TextInput
+            style={{
+               height: 40, borderColor: 'gray', borderWidth: 1, margin: 8
+              }}
+            onChangeText={text => this.setState({ textAttachment: text })}
+          />
+          <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'space-between' }}>
+            <TouchableOpacity
+              style={{ height: 50 }}
+              onPress={() => {
+                    AttachmentsProvider.saveTextAttachment(this.state.textAttachment);
+                    this.dialogComponent.dismiss();
+              }}
+            >
+              <Text style={styles.button}>
+                Save
+              </Text>
+            </TouchableOpacity>
+            <TouchableOpacity
+              style={{ height: 50 }}
+              onPress={() => { this.dialogComponent.dismiss(); }}
+            >
+              <Text style={styles.button}>
+                Cancel
+              </Text>
+            </TouchableOpacity>
+          </View>
+        </View>
+      </DialogComponent>
+    );
+  }
+
+  showFilePicker() {
+    ImagePicker.showImagePicker(null, async (response) => {
+      if (response.didCancel) {
+        console.log('User cancelled image picker');
+      } else if (response.error) {
+        console.log('ImagePicker Error: ', response.error);
+      } else {
+        AttachmentsProvider.saveBinaryAttachment(getFileName(response), response.data, getFileType(response), getFileSize(response));
+        const binaryAttachmentValue = await AttachmentsProvider.getBinaryAttachmentInfo();
+        this.setState({ binaryAttachment: binaryAttachmentValue });
+      }
+    });
+
+    function getFileName(response) {
+      return response.fileName != null ? response.fileName : 'binary.jpeg';
```

Comment:
use `!==` instead of `!=`.

`fileName != null` will false when `fileName` is `null` or `undefined`.